drivers/kw2xrf/kw2xrf_spi.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2015555555555555555555555555555555555555555555555555555555 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_kw2xrf
+ * @{
+ * @file        kw2xrf_spi.c
+ * @brief       Implementation of SPI-functions for the kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "kw2xrf.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_internal.h"
+
+#include "cpu-conf.h"
+#include "periph/gpio.h"
+#include "irq.h"
+
+/* SPI1 connected to the modem (ieee802.15.4 radio) */
+#define KW2XDRF_SPI                   SPI1
+#define KW2XDRF_SPI_CLKEN()           (SIM->SCGC6 |= (SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_CLKDIS()          (SIM->SCGC6 &= ~(SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_IRQ               SPI1_IRQn
+#define KW2XDRF_SPI_IRQ_HANDLER       isr_spi1
+
+void kw2xrf_spi_init(void)
```

Comment:
@jremmert-phytec-iot @jfischer-phytec-iot Although I don't agree with you argumentation I wont NACK it with using this separate SPI implementation. As long as you don't disable interrupts during SPI transactions below.